### PR TITLE
Doc: add a page about security issues

### PR DESCRIPTION
**Pull request by rouault**
Adapted from https://trac.osgeo.org/gdal/wiki/SecurityIssues

### CODE REVIEW

File: doc/source/user/security.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+- Drivers using GDALOpen()/GDALOpenEx()/OGROpen() internally cause other drivers
+  to be used (and their possible flows exploited), without it being obvious at
+  first sight. VRT, MBTiles, KMLSuperOverlay, RasterLite, PCIDSK, PDF, RPFTOC,
+  RS2, WMS, WCS, WFS, OAPIF, OGCAPI, GTI ... are such drivers.
+
+- Drivers depending on downloaded data (HTTP, WMS, WCS, WFS, OAPIF, OGCAPI,
+  STACIT, STACTA, etc.).
+
+- XML based drivers: they might be subject to denial of service by
+  :ref:`​billion laugh-like <https://en.wikipedia.org/wiki/Billion_laughs_attack>`__
```
Comment (dbaston) [inline doc/source/user/security.rst:163]:
```suggestion
  `​billion laugh-like <https://en.wikipedia.org/wiki/Billion_laughs_attack>`__
```

**Comment by rouault** [comment]
@dbaston Thanks for the fixes

File: doc/source/user/security.rst [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+using GDAL. It can also serve as a check-list of items to take care of, for
+developers coding new drivers.
+
+While GDAL maintainers take a number of steps to limit the risk of software
+vulnerabilities, GDAL has a large attack surface, because it deals with data in
+a variety of file formats.
+
+.. note::
+
+    This page is not meant as being a comprehensive cook-book for being "safe" in all
```
Comment (dbaston) [inline doc/source/user/security.rst:17]:
```suggestion
    This page is not intended to be a comprehensive cook-book for being "safe" in all
```

File: doc/source/user/security.rst [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+-------------------------
+
+GDAL network virtual file systems can access credentials needed to grant access
+to remote resources, from the environment or local files. Those secrets are
+stored in RAM using normal mechanisms. Consequently they could potentially be
+recovered by untrusted code that would run in the same process as GDAL code.
+
+The same applies for connection strings to some drivers such as PostgreSQL, MySQL, etc.
+which my include passwords.
+
```
Comment (jratike80) [inline doc/source/user/security.rst:124]: my ->may

Comment (rouault) [reply doc/source/user/security.rst:124]: fixed, thanks

**Pull request merged**